bpf-gpl/tc.c

```diff
@@ -168,79 +169,14 @@ static CALI_BPF_INLINE int calico_tc(struct __sk_buff *skb)
 	}
 
 	/* Copy fields that are needed by downstream programs from the packet to the state. */
-	tc_state_fill_from_iphdr(ctx.state, ctx.ip_header);
+	tc_state_fill_from_iphdr(&ctx);
 
 	/* Parse out the source/dest ports (or type/code for ICMP). */
-	switch (ctx.state->ip_proto) {
```

Comment:
Moved this switch statement to a separate function called "tc_state_fill_from_nextheader(&ctx)" to use it for XDP.

Reply:
Nice.